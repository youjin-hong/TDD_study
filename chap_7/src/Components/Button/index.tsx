import React, {Component} from "react";
import Styled from "styled-components";

interface ContainerProps {
  readonly $backgroundColor: string; // 대소문자 수정
  readonly $hoverColor: string;      // 대소문자 수정
}

interface Props {
  readonly label: string;
  readonly backgroundColor?: string;  // 대소문자 수정
  readonly hoverColor?: string;       // 대소문자 수정
  readonly onClick?: () => void;
}

const Container = Styled.button<ContainerProps>`
  text-align: center;
  background-color: ${(props) => props.$backgroundColor};  // 스타일로 변환됨
  padding: 10px 20px;
  border-radius: 8px;
  cursor: pointer;
  &:hover {
    background-color: ${(props) => props.$hoverColor};  // 스타일로 변환됨
  }
  &:active {
    box-shadow: inset 5px 5px 10px rgba(0, 0, 0, 0.2);
  }
`;

const Label = Styled.div`
  color: #FFFFFF;
  font-size: 16px;
`;

export class Button extends Component<Props> {
  render() {
    const {
      label, backgroundColor = '#304FFE', hoverColor='#1E40FF', onClick
    } = this.props;

    return (
      <Container $backgroundColor={backgroundColor} $hoverColor={hoverColor} onClick={onClick}>
      <label>{label}</label>
      </Container>
    )
  }
}
/**
 export const Button = ({
  label,
  backgroundColor = "#304FFE",   // 대소문자 수정
  hoverColor = "#1E40FF",         // 대소문자 수정
  onClick,
}: Props) => {
  return (
    <Container
      backgroundColor={backgroundColor}  // DOM에 직접 전달되지 않음, 스타일로 적용
      hoverColor={hoverColor}            // DOM에 직접 전달되지 않음, 스타일로 적용
      onClick={onClick}
    >
      <Label>{label}</Label>
    </Container>
  );
};

 */