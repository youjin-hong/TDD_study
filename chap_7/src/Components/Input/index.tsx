import React, { Component } from "react";
import Styled from "styled-components";

interface Props {
  readonly placeholder?: string;
  readonly value?: string;
  readonly onChange?: (text: string) => void;
}

const InputBox = Styled.input`
  flex: 1;
  font-size: 16px;
  padding: 10px 10px;
  border-radius: 8px;
  border: 1px solid #BDBDBD;
  outline: none;
`;

export class Input extends Component<Props> {
  render() {
    const {placeholder, value, onChange} = this.props;
    return (
      <InputBox
      placeholder={placeholder}
      value={value}
      onChange={(event) => {
        if (onChange) {
          onChange(event.target.value)
        }
      }}
      />
    )
  }
}

// export const Input = ({ placeholders, value, onChange }: Props) => {
//   return (
//     <InputBox
//       placeholder={placeholders}
//       value={value}
//       onChange={(event) => {
//         if (typeof onChange === "function") {
//           onChange(event.target.value);
//         }
//       }}
//     />
//   );
// };
