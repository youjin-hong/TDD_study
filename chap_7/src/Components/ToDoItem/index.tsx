import React, { Component } from "react";
import Styled from "styled-components";

import { Button } from "Components/Button";  // Button 컴포넌트 import

interface Props {
  readonly label: string;
  readonly onDelete?: () => void;
}

const Container = Styled.div`
  display: flex;
  border-bottom: 1px solid #BDBDBD;
  align-items: center;
  margin: 10px;
  padding: 10px;
`;

const Label = Styled.div`
  flex: 1;
  font-size: 16px;
  margin-right: 20px;
`;

export class ToDoItem extends Component<Props> {
  render() {
    const { label, onDelete } = this.props;
    return (
      <Container>
        <Label>{label}</Label>
        <Button
          label="삭제"
          backgroundColor="#FF1744"  // $를 사용하여 DOM 속성으로 전달되지 않게 함
          hoverColor="#F01440"       // $를 사용하여 DOM 속성으로 전달되지 않게 함
          onClick={onDelete}
        />
      </Container>
    );
  }
}


// export const ToDoItem = ({ label, onDelete }: Props) => {
//   return (
//     <Container>
//       <Label>{label}</Label>
//       <Button
//         label="삭제"
//         backgroundColor="#FF1744"
//         hoverColor="#F01440"
//         onClick={onDelete}
//       />
//     </Container>
//   );
// };
