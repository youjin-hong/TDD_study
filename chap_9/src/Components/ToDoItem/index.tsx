import React, { Component } from "react";
import Styled from "styled-components";

import { Button } from "Components/Button";  // Button 컴포넌트 import

interface Props {
  readonly label: string;
  readonly onDelete?: () => void;
}

const Container = Styled.div`
  display: flex;
  border-bottom: 1px solid #BDBDBD;
  align-items: center;
  margin: 10px;
  padding: 10px;
`;

const Label = Styled.div`
  flex: 1;
  font-size: 16px;
  margin-right: 20px;
`;


export const ToDoItem = ({ label, onDelete }: Props) => {
  return (
    <Container>
      <Label>{label}</Label>
      <Button
        label="삭제"
        backgroundColor="#FF1744"
        hoverColor="#F01440"
        onClick={onDelete}
      />
    </Container>
  );
};
