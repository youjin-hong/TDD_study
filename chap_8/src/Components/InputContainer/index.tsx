import React, { useContext, useState } from 'react';
import styled from 'styled-components';

import { Button } from 'Components/Button';
import { Input } from 'Components/Input';

import { ToDoListContext } from 'src/Contexts/ToDoList';

const Container = styled.div`
  display: flex;
`;

export const InputContainer = () => {
  const [toDo, setToDo] = useState('');
  const { addToDo } = useContext(ToDoListContext);

  return (
    <Container>
      <Input
        placeholder="할 일을 입력해 주세요."
        value={toDo}
        onChange={setToDo}
      />
      <Button
        label="추가"
        onClick={() => {
          addToDo(toDo);
          setToDo('');
        }}
      />
    </Container>
  );
};
