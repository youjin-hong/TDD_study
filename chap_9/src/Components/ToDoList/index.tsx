import React, { useContext } from 'react';
import styled from 'styled-components';

import { ToDoItem } from 'Components/ToDoItem';

import { ToDoListContext } from 'src/Contexts/ToDoList';

const Container = styled.div`
  width: 350px;
  min-height: 350px;
  height: 400px;
  overflow-y: scroll;
  border: 1px solid #bdbdbd;
  margin-bottom: 20px;
`;

export const ToDoList = () => {
  const { toDoList, deleteToDo } = useContext(ToDoListContext);

  return (
    <Container data-testid="toDoList">
      {toDoList.map((item, index) => (
        <ToDoItem key={item} label={item} onDelete={() => deleteToDo(index)} />
      ))}
    </Container>
  );
};
