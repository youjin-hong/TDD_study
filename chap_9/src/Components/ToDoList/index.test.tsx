import { fireEvent, render, screen } from '@testing-library/react';

import { ToDoListProvider } from 'src/Contexts/ToDoList';
import { ToDoList } from 'Components/ToDoList';

describe('<ToDoList />', () => {
  // 우선 ToDoList 컴포넌트가 화면에 잘 표시되는 지 확인합니다.
  it('renders component correctly', () => {
    const { container } = render(
      <ToDoListProvider>
        <ToDoList />
      </ToDoListProvider>
    );

    const toDoList = screen.getByTestId('toDoList');
    expect(toDoList).toBeInTheDocument();
    expect(toDoList.firstChild).toBeNull();

    expect(container).toMatchSnapshot();
  });

  // 이제 데이터가 있는 경우, 화면에 할 일 목록이 잘 표시되는지 확인합니다.
  it('shows toDo list', () => {
    localStorage.setItem('ToDoList', '["ToDo 1", "ToDo 2", "ToDo 3"]');

    render(
      <ToDoListProvider>
        <ToDoList />
      </ToDoListProvider>
    );

    expect(screen.getByText('ToDo 1')).toBeInTheDocument();
    expect(screen.getByText('ToDo 2')).toBeInTheDocument();
    expect(screen.getByText('ToDo 3')).toBeInTheDocument();
    expect(screen.getAllByText('삭제').length).toBe(3);
  });

  // 마지막으로, 표시된 할 일 목록 데이터를 삭제하는 테스트 명세를 작성합니다.
  it('deletes toDo item', () => {
    localStorage.setItem('ToDoList', '["ToDo 1", "ToDo 2", "ToDo 3"]');

    render(
      <ToDoListProvider>
        <ToDoList />
      </ToDoListProvider>
    );

    const toDoItem = screen.getByText('ToDo 2');
    expect(toDoItem).toBeInTheDocument();
    fireEvent.click(toDoItem.nextElementSibling as HTMLElement);
    expect(toDoItem).not.toBeInTheDocument();
    expect(
      JSON.parse(localStorage.getItem('ToDoList') as string)
    ).not.toContain('ToDo 2');
  });
});
