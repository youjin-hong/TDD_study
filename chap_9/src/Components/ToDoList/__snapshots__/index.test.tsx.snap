// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<ToDoList /> > renders component correctly 1`] = `
<div>
  <div
    class="sc-dTvVRJ AQmon"
    data-testid="toDoList"
  />
</div>
`;
